components/avatar/avatar.vue

```diff
@@ -252,107 +296,77 @@ export default {
       // TODO: Group only supports xs size for now. Remove this when we support other sizes.
       return this.group ? 'xs' : this.size;
     },
-  },
 
-  mounted () {
-    this.init();
+    showImage () {
+      return this.imageLoadedSuccessfully !== false && this.imageSrc;
+    },
   },
 
-  updated () {
-    this.init();
+  watch: {
+    fullName: {
+      immediate: true,
+      handler (newName) {
+        this.formatInitials(newName);
+      },
+    },
+
+    imageSrc: {
+      immediate: true,
+      async handler (newSrc) {
+        if (!newSrc) return;
+
+        this.validateProps();
+
+        this.imageLoadedSuccessfully = null;
+        await this.$nextTick();
+        this.setImageListeners();
```

Comment:
It feels a bit flaky to do `nextTick` in an immediate watcher to wait for the UI to update. I think it would be much safer to change this to a non-immediate watcher and put the code within it into a method and call this method here and in `mounted()`